src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -1,42 +1,50 @@
 import { div, h, span } from 'react-hyperscript-helpers'
 import * as breadcrumbs from 'src/components/breadcrumbs'
-import { buttonPrimary } from 'src/components/common'
+import { buttonPrimary, spinnerOverlay } from 'src/components/common'
 import { centeredSpinner } from 'src/components/icons'
 import Modal from 'src/components/Modal'
 import { Rawls } from 'src/libs/ajax'
 import * as Nav from 'src/libs/nav'
+import * as StateHistory from 'src/libs/state-history'
 import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
 import { Component } from 'src/libs/wrapped-components'
 import WorkspaceContainer from 'src/pages/workspaces/workspace/WorkspaceContainer'
 
 
 export default class WorkspaceDashboard extends Component {
+  constructor(props) {
+    super(props)
+    this.state = StateHistory.get()
+  }
+
   refresh() {
     const { namespace, name } = this.props
 
-    this.setState({ workspace: undefined, workspaceFailure: undefined })
     Rawls.workspace(namespace, name).details().then(
-      workspace => this.setState({ workspace }),
+      workspace => this.setState({ freshData: true, workspace }),
       workspaceFailure => this.setState({ workspaceFailure })
     )
   }
 
   render() {
-    const { modal, workspace, workspaceFailure } = this.state
+    const { freshData, modal, workspace, workspaceFailure } = this.state
     const { namespace, name } = this.props
 
     return h(WorkspaceContainer,
       {
-        namespace, name, refresh: () => this.refresh(),
+        namespace, name, refresh: () => {
+          this.setState({ freshData: false })
```

Comment:
Seeing this makes me wonder: are we actually just tracking whether we're currently fetching data? Would this flag be better named `isLoading`, and tied to the underlying ajax flow?